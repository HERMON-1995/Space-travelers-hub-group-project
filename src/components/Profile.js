import React from 'react';
import { useSelector } from 'react-redux';

const Profile = () => {
  const reservedRockts = useSelector((state) => state.rocketsReducer.filter((rck) => rck.reserved));
  const misprfo = useSelector((state) => state.missionReducer.missionList
    .filter((mis) => mis.missionValue));

  return (
    <div className="row">
      <div className=" col-6">
        <h3 className="ms-5 mt-3">My Rockets</h3>
        <ul className="list-group ms-5">
          {reservedRockts.length ? reservedRockts.map((rocket) => <li key={rocket.id} className="list-group-item">{rocket.name}</li>) : <li className="list-group-item">No Reserved Rockets</li> }
        </ul>
      </div>
      <div className="col-6">
        <h3 className="ms-5 mt-3">My Missions</h3>
        <ul className="list-group ms-5">
          {misprfo.length ? misprfo.map((miss) => <li key={miss.id} className="list-group-item">{miss.missionName}</li>) : <li className="list-group-item">No Mission Reserved</li> }
        </ul>
      </div>
    </div>
  );
};

export default Profile;
