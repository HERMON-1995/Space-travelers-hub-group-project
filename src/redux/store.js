import thunk from 'redux-thunk';
import { legacy_createStore as createStore, combineReducers, applyMiddleware } from '@reduxjs/toolkit';
import rockets from './rocketsSlice';
import mission from './missionsSlice';

const reducers = combineReducers({
  rocketsReducer: rockets,
  missionReducer: mission,
});

const store = createStore(reducers, applyMiddleware(thunk));

export default store;
