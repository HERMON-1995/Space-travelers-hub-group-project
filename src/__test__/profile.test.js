import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Profile from '../components/Profile';
import store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Profile />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});