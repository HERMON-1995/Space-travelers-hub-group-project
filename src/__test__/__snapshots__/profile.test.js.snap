// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="row"
>
  <div
    className=" col-6"
  >
    <h3
      className="ms-5 mt-3"
    >
      My Rockets
    </h3>
    <ul
      className="list-group ms-5"
    >
      <li
        className="list-group-item"
      >
        No Reserved Rockets
      </li>
    </ul>
  </div>
  <div
    className="col-6"
  >
    <h3
      className="ms-5 mt-3"
    >
      My Missions
    </h3>
    <ul
      className="list-group ms-5"
    >
      <li
        className="list-group-item"
      >
        No Mission Reserved
      </li>
    </ul>
  </div>
</div>
`;
